### PR TITLE
check anagram done and ready for review

### PR DESCRIPTION
**Pull request by TekaMesfinAbel**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/check_anagram.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"""
```
Comment (ViktoriyaHaiduk) [inline solutions/check_anagram.py:1]: Please add header due to Evan's example:
https://github.com/MIT-Emerging-Talent/ET6-Programming-With-Python/tree/main/3_documenting_and_testing/examples

**Review by ViktoriyaHaiduk — DISMISSED** [review]
Great job on the implementation! Your function is well-structured, and the documentation clearly explains how it works. Overall, the code is clean, concise, and easy to understand. Keep up the good work!

File: solutions/check_anagram.py (hunk 1 above)
Comment (TekaMesfinAbel) [reply solutions/check_anagram.py:1]: Thank you viktoriya. I will do that.

**Review by ViktoriyaHaiduk — APPROVED** [review]
Great job on the implementation!

File: solutions/check_anagram.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    Args:
+        string1 (str): The first string to compare.
+        string2 (str): The second string to compare.
+
+    Returns:
+        bool: True if the strings are anagrams, False otherwise.
+
+    Raises:
+        TypeError: If either argument is not a string.
```
Comment (GaiSamuel) [inline solutions/check_anagram.py:23]: Raises Attribute error not type error according to the test provided.

File: solutions/tests/test_check_anagram.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        """Test with strings containing special characters."""
+        self.assertFalse(is_anagram("evil!", "vile"))
+
+    def test_anagram_numeric_values(self):
+        """Test with numeric strings."""
+        self.assertFalse(is_anagram("123", "321a"))
+
+    def test_invalid_input_not_string(self):
+        """Test for TypeError when inputs are not strings."""
+        with self.assertRaises(TypeError):
```
Comment (GaiSamuel) [inline solutions/tests/test_check_anagram.py:60]: Raises attribute error not type error.

File: solutions/tests/test_check_anagram.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            is_anagram(["list"], "silent")
+        with self.assertRaises(TypeError):
```
Comment (GaiSamuel) [inline solutions/tests/test_check_anagram.py:62]: Raises attribute error not type error.

**Review by GaiSamuel — CHANGES_REQUESTED** [review]
Looks good generally. Just a few corrections added.

File: solutions/check_anagram.py (hunk 2 above)
Comment (TekaMesfinAbel) [reply solutions/check_anagram.py:23]: Ohh yeah. Thanks for catching that.

**Review by ViktoriyaHaiduk — APPROVED** [review]
Good job. Everything is according to checklist.

**Review by GaiSamuel — APPROVED** [review]
Looks good! Great work Abel.

**Review by ViktoriyaHaiduk — APPROVED** [review]
Great job on the implementation! Your function is well-structured, and the documentation clearly explains how it works. Overall, the code is clean, concise, and easy to understand. Keep up the good work!
